packages/workflow/src/workflow.ts

```diff
@@ -1133,3 +1134,52 @@ export function setHandler<Ret, Args extends any[], T extends SignalDefinition<A
     throw new TypeError(`Invalid definition type: ${(def as any).type}`);
   }
 }
+
+/**
+ * Updates this Workflow's Search Attributes by merging the provided `searchAttributes` with the existing Search
+ * Attributes, `workflowInfo().searchAttributes`.
+ *
+ * For example, this Workflow code:
+ *
+ * ```ts
+ * upsertSearchAttributes({
+ *   CustomIntField: [1],
+ *   CustomBoolField: [true]
+ * });
+ * upsertSearchAttributes({
+ *   CustomIntField: [2, 3],
+ *   CustomKeywordField: ['durable code']
+ * });
+ * ```
+ *
+ * would result in the Workflow having these Search Attributes:
+ *
+ * ```ts
+ * {
+ *   CustomIntField: [2, 3],
+ *   CustomBoolField: [true],
+ *   CustomKeywordField: ['durable code']
+ * }
+ * ```
+ *
+ * @param searchAttributes The Record to merge. Use a value of `[]` to clear a Search Attribute.
+ */
+export function upsertSearchAttributes(searchAttributes: Record<string, SearchAttributeValue>): void {
+  if (!state.info) {
+    throw new IllegalStateError('`state.info` should be defined');
+  }
+
+  const mergedSearchAttributes = { ...state.info.searchAttributes, ...searchAttributes };
+  if (!mergedSearchAttributes) {
+    throw new Error('searchAttributes must be a non-null Record<string, SearchAttributeValue>');
+  }
+
+  state.pushCommand({
+    upsertWorkflowSearchAttributesCommandAttributes: {
```

Comment:
@Sushisource this should be called `upsertSearchAttributes` IMHO, we definitely don't need the `CommandAttributes` suffix, this is the only place where we use this pattern.

Reply:
Sure, can do that.

Reply:
https://github.com/temporalio/sdk-core/issues/332